### PR TITLE
City State models created to populate sign up dropdowns, Closes #11, #37 and #38

### PR DESCRIPTION
**Pull request by cluhring**
I added coffeescript/jquery to dynamically populate the state & city fields on our company form.
I created city and state models (states have many cities).  I made a few rake tasks to populate your local city and state database.  
- rake populate_form runs
  - rake states & rake cities

I also installed 'dalli' and configured a cache_store in config/environments/dev to cache the cities in the company signup form / not hit the database everytime you visit the signup page.
Still need to reverse city order in form & add a couple tests.

### CODE REVIEW

File: logfile [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (cluhring) [inline logfile]: just added logfile to my global gitignore, sorry for including.  possible to delete file?

File: app/views/application/_signup_form.html.erb [hunk 2]
```diff
@@ -17,11 +17,13 @@ more about TechHire, and a passion to find and hire talented employees in your c
...
     <div class="form-group">
       <%= f.label(:state, "State") %>
-      <%= f.select :state, options_for_select(state_helper), placeholder: "AL" %>
+      <%= f.select :state, options_for_select(state_helper) %>
     </div>
     <div class="form-group">
       <%= f.label(:city, "City") %>
-      <%= f.select :city, options_for_select(city_helper), placeholder: "Please select..." %><br>
+      <% cache @cities do %>
+        <%= f.grouped_collection_select :city, State.order(:id), :usa_cities, :abbr, :id, :name %><br>
```
Comment (oorion) [inline app/views/application/_signup_form.html.erb]: Could you pull the State.order(:id) logic out and put it in the model?

File: spec/features/guest_user_spec.rb [hunk 3]
```diff
@@ -62,5 +90,14 @@
       expect(current_path).to eq(companies_path)
       expect(page.find('.errors')).to have_content("Please try again!")
     end
+
+    it "can see dynamically generated cities" do
```
Comment (oorion) [inline spec/features/guest_user_spec.rb:94]: Nice test! :sparkle:

File: app/views/application/_signup_form.html.erb (hunk 2 above)
Comment (cluhring) [inline app/views/application/_signup_form.html.erb]: State.all ok?

File: spec/models/usa_city_spec.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+require 'rails_helper'
+
+RSpec.describe UsaCity, type: :model do
+  let (:city1) {
+    UsaCity.create(
+                   name: "Denver",
+               state_id: 14
```
Comment (laurawhalin) [inline spec/models/usa_city_spec.rb]: Can you nudge this over so the alignment looks better. Sorry, I know that's super nitpicky :cactus:

File: spec/features/guest_user_spec.rb [hunk 5]
```diff
@@ -35,7 +63,7 @@
         fill_in 'company[organization]', with: "Google"
         fill_in 'company[title]', with: "RoR Developer"
         select('CO', from: 'company[state]')
-        select('Fort Collins', from: 'company[city]')
+        select('Denver', from: 'company[city]')
```
Comment (laurawhalin) [inline spec/features/guest_user_spec.rb:66]: Denver is a way better choice than Fort Collins :wink:

**Comment by scottalexandra** [comment]
Nice Work Chris!

File: config/environments/development.rb [hunk 6]
```diff
@@ -2,7 +2,10 @@
   config.cache_classes = false
   config.eager_load = false
   config.consider_all_requests_local       = true
-  config.action_controller.perform_caching = false
+  # Might want to look into database caching, per Horace
+  config.action_controller.perform_caching = true
+  config.cache_store = :dalli_store
+  # Don't care if the mailer can't send.
```
Comment (laurawhalin) [inline config/environments/development.rb:8]: What do you want to do about these comments? Take action or delete them?

**Pull request merged**